packages/eslint-plugin/src/rules/ban-types.ts

```diff
@@ -96,8 +96,13 @@ export default util.createRule<Options, MessageIds>({
           fixWith: 'number',
         },
         Object: {
-          message: 'Use Record<string, any> instead',
-          fixWith: 'Record<string, any>',
+          message:
+            'The Object type is mostly the same as unknown, you probably want Record<string, unknown> instead'
+        },
+        object: {
+          message:
+            'The object type is hard to use, use Record<string, unknown> instead',
+          fixWith: 'Record<string, unknown>'
```

Comment:
Any reason to recommend `Record<string, unknown>` over `{ [key: string]: unknown };`? Would be good to document this in the rule docs.

Reply:
IMO it's more readable, and it looks closer to a named type, which this rule is built around.
An indexed object type feels more jarring in a fix scenario:
```ts
const x: object = {};
// feels nicer fixed to
const x: Record<string, unknown> = {};
// instead of
const x: {[key: string]: unknown} = {};
```

Reply:
Then I would document that as a reasoning in the docs - That you picked `Record` as it's more readable, but that `{[key: string]: unknown}` would work fine too.